src/main/java/io/smartcat/migration/CassandraVersioner.java

```diff
@@ -3,86 +3,50 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.datastax.driver.core.ConsistencyLevel;
-import com.datastax.driver.core.ResultSet;
 import com.datastax.driver.core.Row;
 import com.datastax.driver.core.Session;
-import com.datastax.driver.core.Statement;
-import com.datastax.driver.core.querybuilder.QueryBuilder;
+
+import java.util.Optional;
+
 
 /**
- * Class responsible for version management.
+ * Saves migrations with an Integer value.
  */
-public class CassandraVersioner {
+public class CassandraVersioner
+    extends AbstractVersioner<Integer> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CassandraVersioner.class);
 
-    private static final String SCHEMA_VERSION_CF = "schema_version";
-    private static final String TYPE = "type";
-    private static final String VERSION = "version";
-    private static final String TIMESTAMP = "ts";
-    private static final String DESCRIPTION = "description";
-
-    private static final String CREATE_SCHEMA_VERSION_CQL = String.format("CREATE TABLE IF NOT EXISTS %s (",
-            SCHEMA_VERSION_CF)
-            + String.format("%s text,", TYPE)
-            + String.format("%s int,", VERSION)
-            + String.format("%s bigint,", TIMESTAMP)
-            + String.format("%s text,", DESCRIPTION)
-            + String.format("PRIMARY KEY (%s, %s)", TYPE, VERSION)
-            + String.format(")  WITH CLUSTERING ORDER BY (%s DESC)", VERSION) + " AND COMMENT='Schema version';";
-
-    private final Session session;
 
     /**
-     * Create Cassandra versioner for active session.
-     * @param session Active Cassandra session
+     * Create Cassandra versioner.
      */
-    public CassandraVersioner(final Session session) {
-        this.session = session;
+    public CassandraVersioner() {
```

Comment:
This class does have some changed contracts - I can work in their preservation if you think this is a non-internal class.